### DISCUSSION TITLE
How to use `$VISUAL`?

### DISCUSSION THREAD

**Discussion by eric-hansen** [dcampos/nvim-snippy#98] [Q&A] [answered]
I'm new to snippet engines and like the minimalism of this project.

The linked repo for snippet samples (honza/vim-snippets) uses `$VISUAL` a lot, and I was wondering how to make that work exactly?

Since using `nvim-cmp` it's not clear to me how to both insert the trigger text and be in visual mode at the same time.

**Comment by dcampos** [comment, marked as answer]
You need to select some text in visual mode, then press the key you've mapped to the `cut_text` action, followed the trigger for the snippet and expand it. The text you selected will replace the `$VISUAL` placeholder in the snippet.

In normal mode, you can also use a mapping for `cut_text` to do the same thing. Let's say the mapping is `g<Tab>` and you want to use text within brackets, simply press `g<Tab>ib`, enter the trigger and expand the snippet with the `$VISUAL` placeholder.

**Reply by eric-hansen** [reply]
Thanks!  I couldn't figure out what the objective was for `cut_text`, now it makes sense.
